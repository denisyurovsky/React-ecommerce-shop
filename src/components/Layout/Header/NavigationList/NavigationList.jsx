import React from 'react';
import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';

import { USER_ROLE } from '../../../../helpers/constants/constants';
import { pathNames } from '../../../../helpers/constants/pathNames/pathNames';
import { getUserRole } from '../../../../store/user/userSlice';

import styles from './NavigationList.module.scss';

export const NavigationList = () => {
  const userRole = useSelector(getUserRole);

  const navItems = [
    { title: 'home', path: '/' },
    { title: 'products', path: pathNames.PRODUCTS },
  ];

  if (userRole === USER_ROLE.ADMIN || userRole === USER_ROLE.SELLER) {
    navItems.push({ title: 'admin', path: pathNames.ADMIN });
  }

  return (
    <ul data-testid="header-navigation-list" className={styles.list}>
      {navItems.map((navItem, i) => (
        <NavLink
          end
          to={navItem.path}
          key={`navItem${i}`}
          className={styles.link}
        >
          {navItem.title}
        </NavLink>
      ))}
    </ul>
  );
};
