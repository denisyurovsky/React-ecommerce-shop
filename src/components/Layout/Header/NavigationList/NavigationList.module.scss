@import "../../../../assets/styles/colors.scss";

.list {
  list-style: none;
  padding: 0;
  margin: 0;
  display: flex;
  flex-direction: row;
  width: min-content;
}

.link {
  text-decoration: none;
  text-transform: uppercase;
  padding: 20px;
  color: $greyLight;
  background-color: transparent;
  font-size: 16px;

  &:hover {
    cursor: pointer;
    background: rgba($greyLight, .3); // stylelint-disable-line
  }
}

.active {
  background: rgba($greyLight, .3); // stylelint-disable-line
}