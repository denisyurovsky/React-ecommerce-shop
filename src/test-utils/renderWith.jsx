import { configureStore } from '@reduxjs/toolkit';
import { render as rtlRender } from '@testing-library/react';
import PropTypes from 'prop-types';
import React from 'react';
import { Provider } from 'react-redux';
import { ToastContainer } from 'react-toastify';

import cartReducer from '../store/cart/cartSlice';
import categoriesReducer from '../store/categories/categoriesSlice';
import feedbackReducer from '../store/feedback/feedbackSlice';
import productsReducer from '../store/products/productsSlice';
import userReducer from '../store/user/userSlice';

import RouterConnected from './RouterConnected';

function render(
  component,
  preloadedState = {},
  path = '/',
  initialPaths = ['/']
) {
  const store = configureStore({
    reducer: {
      cart: cartReducer,
      products: productsReducer,
      categories: categoriesReducer,
      user: userReducer,
      feedback: feedbackReducer,
    },
    preloadedState,
  });

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <RouterConnected
          component={children}
          path={path}
          initialPaths={initialPaths}
        />
        <ToastContainer />
      </Provider>
    );
  }

  Wrapper.propTypes = {
    children: PropTypes.object,
  };

  return rtlRender(component, { wrapper: Wrapper });
}

export * from '@testing-library/react';
export default render;
