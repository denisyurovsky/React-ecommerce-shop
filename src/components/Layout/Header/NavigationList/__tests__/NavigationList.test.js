import React from 'react';

import renderWith from '../../../../../test-utils/renderWith';
import { NavigationList } from '../NavigationList';

describe('NavigationList component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot', () => {
      const { asFragment } = renderWith(<NavigationList />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
