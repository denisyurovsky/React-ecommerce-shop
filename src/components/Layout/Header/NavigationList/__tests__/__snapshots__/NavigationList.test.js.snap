// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationList component snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <ul
    class="list"
    data-testid="header-navigation-list"
  >
    <a
      aria-current="page"
      class="link active"
      href="/"
    >
      home
    </a>
    <a
      class="link"
      href="/products"
    >
      products
    </a>
  </ul>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
